File: app/src/main/java/technology/dubaileading/maccessemployee/BaseApplication.java
package technology.dubaileading.maccessemployee;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatDelegate;

public class BaseApplication extends Application implements Application.ActivityLifecycleCallbacks{
    private static BaseApplication instance;

    private static Activity currentActivity;

//    BottomNavigationView view;

    //private FirebaseNotificationContract.Presenter presenter;

    public BaseApplication() {
        instance = this;
    }

    public Activity getCurrentActivity(){
        return currentActivity;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
        registerActivityLifecycleCallbacks(this);

//        view.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
//            @Override
//            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
//                return false;
//            }
//        });
    }


    public static synchronized BaseApplication getInstance() {
        return instance;
    }

    @Override
    public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
        currentActivity = activity;
    }

    @Override
    public void onActivityStarted(@NonNull Activity activity) {
        currentActivity = activity;
    }

    @Override
    public void onActivityResumed(@NonNull Activity activity) {
        currentActivity = activity;
    }

    @Override
    public void onActivityPaused(@NonNull Activity activity) {
        //currentActivity = null;
    }

    @Override
    public void onActivityStopped(@NonNull Activity activity) {
        //currentActivity = null;
    }

    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(@NonNull Activity activity) {
        //currentActivity = null;
    }
}

File: app/src/main/java/technology/dubaileading/maccessemployee/BaseApplication.kt
package technology.dubaileading.maccessemployee

import android.app.Activity
import android.app.Application
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle
import androidx.appcompat.app.AppCompatDelegate
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class BaseApplication : Application(), ActivityLifecycleCallbacks {
    init {
        instance = this
    }

    override fun onCreate() {
        super.onCreate()
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
        registerActivityLifecycleCallbacks(this)

    }

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
        Companion.currentActivity = activity
    }

    override fun onActivityStarted(activity: Activity) {
        Companion.currentActivity = activity
    }

    override fun onActivityResumed(activity: Activity) {
        Companion.currentActivity = activity
    }

    override fun onActivityPaused(activity: Activity) {
        //currentActivity = null;
    }

    override fun onActivityStopped(activity: Activity) {
        //currentActivity = null;
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
    override fun onActivityDestroyed(activity: Activity) {
        //currentActivity = null;
    }

    companion object {
        @get:Synchronized
        lateinit var instance: BaseApplication
            private set
        private var currentActivity: Activity? = null
    }
}